db/user.go

```diff
@@ -221,3 +222,71 @@ func SyncUsers(course *model.Course, newUsers map[string]*model.User,
 
     return syncResult, nil;
 }
+
+// ResolveUsers is a helper function to maps string representations of roles and * (all roles) to the emails for users with those roles.
+// The function takes an optional course and a list of strings, containing emails, roles, and * as input and returns an alphanumerically sorted slice of lowercase emails without duplicates.
+// A fatal error occurs if a course cannot be opened and the input contains roles or a *.
+// A warning will be logged for every invalid role.
```

Comment:
You can leave out this comment.
We normally don't comment on up-front about every warning that we may give,
and I see no reason for an exception here.

Note that this is for a very important version, not just to be lazy.
We want ever comment to be correct and stay correct (hence all the changes on this round of the PR).
This means that every comment has a maintenance burden associated with it,
and comments like that have a high burden (can easily change if anyone else modifies the function).